pkg/config/deployment.go

```diff
@@ -55,6 +58,19 @@ type DeploymentPlanner struct {
 	AlwaysUsePipeline bool `json:"alwaysUsePipeline"`
 }
 
+type Trigger struct {
+	// Configuration in case changes on commit are found.
+	OnCommit OnCommitConfig `json:"onCommit"`
+}
+
+type OnCommitConfig struct {
+	// Control trigger new deployment on Git change or not.
+	Disable bool `json:"disable,omitempty"`
```

Comment:
"Disabled" because we are using the same format at other places.

Reply:
nit: Do we need the `Config` suffix?

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2714/commits/b9bc4cc9de939e2a3fde2bb799a164c0f51c87cc